source/solvers/fluid_dynamics_matrix_free_operators.cc

```diff
@@ -410,8 +441,23 @@ NavierStokesOperatorBase<dim, number>::vmult(VectorType       &dst,
         .local_element(edge_constrained_indices[i]) = 0.;
     }
 
+#if DEAL_II_VERSION_GTE(9, 6, 0)
```

Comment:
Is there somewhere an assert that weak dbc is not working for older versions of deal.II?

Reply:
Yeah I added an assertion elsewhere.  The code will crash now in Deal.ii 9.5 or below